spec.md

```diff
@@ -2734,6 +2780,270 @@ message NodeExpandVolumeResponse {
+
+#### `CreateVolumeGroupSnapshot`
+
+**ALPHA FEATURE**
+
+A Group Controller Plugin MUST implement this RPC call if it has `CREATE_DELETE_GET_VOLUME_GROUP_SNAPSHOT` controller capability.
+This RPC will be called by the CO to create a new volume group snapshot from a list of source volumes on behalf of a user.
+
+The purpose of this call is to request the creation of a multi-volume snapshot. Group snapshots can be created from the existing list of volumes. Note that calls to this function MUST be idempotent - the function may be called multiple times for the same name - the group snapshot must only be created once.
+
+If a group snapshot corresponding to the specified group snapshot `name` is successfully created (meaning all snapshots associated with the group are successfully cut), the Plugin MUST reply `0 OK` with the corresponding `CreateVolumeGroupSnapshotResponse`.
+
+If an error occurs before a group snapshot is cut, `CreateVolumeGroupSnapshot` SHOULD return a corresponding gRPC error code that reflects the error condition.
+
+`CreateVolumeGroupSnapshot` SHOULD return `0 OK` after all the snapshots are cut regardless of whether post processing such as uploading is complete or not.
+
+CO SHOULD check the `ready_to_use` boolean of individual snapshots that are members of the group snapshot and only treat the group snapshot as ready to use when all snapshots have been "processed" and is ready to use to create new volumes from those snapshots.
+
+An individual snapshot MAY be used as the source to provision a new volume.
+
+In VolumeGroupSnapshot message, both snapshots and group_snapshot_id are required fields.
+
+If an error occurs before all the individual snapshots are created when creating a group snapshot of multiple volumes and a group_snapshot_id is not yet available for CO to do clean up, SP SHOULD do clean up and make sure no snapshots are leaked.
+
+```protobuf
+message CreateVolumeGroupSnapshotRequest {
+  option (alpha_message) = true;
+
+  // The suggested name for the group snapshot. This field is REQUIRED
+  // for idempotency.
+  // Any Unicode string that conforms to the length limit is allowed
+  // except those containing the following banned characters:
+  // U+0000-U+0008, U+000B, U+000C, U+000E-U+001F, U+007F-U+009F.
+  // (These are control characters other than commonly used whitespace.)
+  string name = 1;
+
+  // volume ids of the source volumes to be snapshotted together.
+  // This field is REQUIRED.
+  repeated string source_volume_ids = 2;
+
+  // Secrets required by plugin to complete
+  // ControllerCreateVolumeGroupSnapshot request.
+  // This field is OPTIONAL. Refer to the `Secrets Requirements`
+  // section on how to use this field.
+  // The secrets provided in this field SHOULD be the same as
+  // the secrets provided in ControllerDeleteVolumeGroupSnapshot
+  // and ControllerGetVolumeGroupSnapshot requests for the same
+  // group snapshot unless if secrets are rotated after the
```

Comment:
re: secret rotation - why is special language used for the snapshot APIs vs. elsewhere? IMO we should be consistent w/ respect to how we describe expectations around secret longevity. otherwise this looks like a special case (and I do not believe that is the intent here)

Reply:
Lines 2870-2873 are added to address a review comment, mainly to make it clear that the secrets for the same VolumeGroup should be the same for all operations. If you are fine with that part, maybe we can use similar language for other secrets throughout the doc in a separate PR.  I can drop the secrets rotation part if it adds confusion.

Reply:
Yeah we should be consistent about recommendations for secret longevity, but it doesn't make sense to do all that work in this PR.

Reply:
Changed to this:
```
// The secrets provided in this field SHOULD be the same for
// all group snapshot operations on the same group snapshot.
```